src/contexts/AppContext.js

```diff
@@ -23,8 +24,33 @@ const baseState = {
 const AppProvider = (props) => {
 
     const [state, dispatch] = useReducer(appReducer, {}, () => {
-        const localData = localStorage.getItem('app');
-        return localData ? {...baseState, ...JSON.parse(localData)} : baseState;
+        const appString = localStorage.getItem('app');
+        let localData;
+
+        if (appString) {
+            localData = JSON.parse(appString);
+        } else {
+            localData = {};
+        }
+
+        /*
+         * Override connection properties with those specified in the query
+         * string (if any) and default the SIP and websocket URIs.
+         */
+        new URLSearchParams(window.location.search).forEach((value, key) => {
+            localData[key] = value;
```

Comment:
A tiny tiny thing.... I'd probably only set it into localData if the key exists in baseState. That way you cant pass 100 things in the URL and have them all appear in the state of the app